web-ui/src/streaming/inspection/InspectionTable.tsx

```diff
@@ -0,0 +1,305 @@
+          '/egress/' +
+          name +
+          '?format=csv&query=quantiles&mode=snapshot'
+      )
+      quantileLoader(url, setQuantiles, relation, controller).then(
+        () => {
+          // nothing to do here, the tableUpdater will update the table
+        },
+        (error: any) => {
+          if (error.name != 'AbortError') {
+            throwError(error)
+          } else {
+            // The AbortError is expected when we leave the view
+            // (controller.abort() below will trigger it)
+            // -- so nothing to do here
+          }
+        }
+      )
+
+      return () => {
+        // If we leave the view, we abort the fetch request otherwise it remains
+        // active (the browser and backend only keeps a limited number of active
+        // requests and we don't want to exhaust that limit)
+        controller.abort()
+      }
+    }
+  }, [pipeline, name, relation, quantileLoader, rows, throwError, quantiles])
+
+  // If the user presses the back or next button, we update the neighborhood
+  // which will trigger update the rows in the table.
+  const handlePaginationModelChange = useCallback(
+    (newPaginationModel: GridPaginationModel) => {
+      if (!relation) {
+        return
+      }
+      if (paginationModel.page > newPaginationModel.page) {
+        // Going backward, we can only go backward as long as we have at least
+        // one item with a genId < 0
+        const minRow = rows.find((row: any) => row.genId < 0)
+        if (!minRow) {
+          pushMessage({
+            message: `You reached the beginning of the ${relation.name} relation.`,
+            key: new Date().getTime(),
+            color: 'info'
+          })
+          return
+        }
+        setRows([])
+        setNeighborhood({
+          ...DEFAULT_NEIGHBORHOOD,
+          anchor: objectToAnchor(relation, minRow)
+        })
+      } else {
+        if (rows[rows.length - 1].genId === PAGE_SIZE) {
+          // Going forward, we can only go forward if we have one more item than
+          // what's being displayed in the table (genId == PAGE_SIZE)
+          setRows([])
+          setNeighborhood({
+            ...DEFAULT_NEIGHBORHOOD,
+            anchor: objectToAnchor(relation, rows[rows.length - 1])
+          })
+        } else {
+          pushMessage({
+            message: `You reached the end of the ${relation.name} relation.`,
+            key: new Date().getTime(),
+            color: 'info'
+          })
+          return
+        }
+      }
+      setQuantile(0)
+      setPaginationModel(newPaginationModel)
+    },
+    [rows, relation, paginationModel, pushMessage]
+  )
+
+  // If we selected a quantile from the slider, we also update the neighborhood.
+  function onQuantileSelected(event: Event, value: number | number[]): void {
+    // `value` is always a number since we only have one slider and we're not
+    // setting a range.
+    const quantileNumber = value as number
```

Comment:
I changed it to percentile in the user-visible part. For the API part I'll keep the names as quantile as it can deal with arbitrary ranges and in the backend API docs we also call it quantile.